### ISSUE TITLE
[Docs] Fix up caching docs as per user feedback

### ISSUE THREAD

**Issue by ppiegaze** [flyteorg/flyte#3249]
### Description

In [this Slack thread](https://flyte-org.slack.com/archives/CP2HDHKE1/p1674155785990949?thread_ts=1672163653.068519&cid=CP2HDHKE1) we find the following advice (we should act on it):

Niels Bantilan
[@Sidharth (Sid)](https://flyte-org.slack.com/team/U04CEE0LCTY)
 yes our [flyte.org](http://flyte.org/) website revamp (coming soon!) should feature caching a lot more prominently. Out of curiosity, do you find the [current docs on caching](https://docs.flyte.org/projects/cookbook/en/latest/auto/core/flyte_basics/task_cache.html#sphx-glr-auto-core-flyte-basics-task-cache-py) clear and understandable?

Sidharth(Sid)
  [18 hours ago](https://flyte-org.slack.com/archives/CP2HDHKE1/p1674155785990949?thread_ts=1672163653.068519&cid=CP2HDHKE1)
I feel it has everything a person needs to understand Caching, but it needs a thorough read, maybe even a few times for a complete new comer to workflow tools.
Also, terms related to caching such as "task Signature" could be simpler in my opinion. For example "Any changes to task Function".
Prefect V1 docs for caching - https://docs-v1.prefect.io/core/concepts/persistence.html#output-caching-based-on-a-file-target
Its pretty old, but has similar concepts explained in simple sentences.
Im just sharing it so that I can show you what kind of understanding fits a crowd who are new to these tools. But I do wish the above prefect page to be more technical. Closer to flyte's way of explaining.
Basically a middle ground will be nice.
Also in Flyte current stable docs, I would wish for more details on "local cache storage" and "remote cache storage". The above prefect v1 doc gives some insight to it.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by sid-geospoc** [comment]
Few more points apart from those I had mentioned in thread -
1. the 14min video part of the current stable docs is too long, rather a short 1/2 min video that goes through an example would be great. The video should leave enough room for thought, that the user then reads the technical docs to get complete picture.
2. The new example in video could be some data science specific cache usage, and if possible, cross workflow caching usage, to show the true potential of caching in a DS/ML scenario.
3. Both local and remote cache usage would be great, including registering of files.

Thanks @ppiegaze for creating this thread.

**Comment by sid-geospoc** [comment]
Finally one small detail which could be changed is the "browser tab icon" ie., Favicon,  for Flyte Docs and Flyte UI.
This helps when multiple flyte sites such as Blogs, Docs, and UI is opened up in a browser, along with multiple other tabs. :)

**Comment by 10sharmashivam** [comment]
Hey,

I would like to work on this issue. I’ll focus on simplifying the existing caching documentation to make it more accessible for a wider audience, especially those who might not be familiar with advanced technical terms.

I will be:

	1.	Simplifying Technical Terms: I will replace complex or technical terms (e.g., “task signature,” “cache_serialize”, "cache_ignore_input_vars") with simpler explanations and real-world examples to improve readability.
	2.	Restructuring for Clarity: I’ll ensure the structure of the documentation flows logically, making it easier for users to follow. This includes improving section titles and breaking down longer explanations into more digestible parts.
	3.	Short Video Demo: I can also create a short 1-2 minute video that provides a brief overview of caching and shows how to enable/disable it in Flyte. This would help users quickly grasp the concept. (If required).

Since simplifying documentation can be subjective, I’m open to feedback and suggestions to ensure the changes are helpful for the community.

Thanks! #take

**Comment by davidmirror-ops** [comment]
>While reviewing the documentation again, I got a question: Are checkpoints and targets part of Flyte’s caching mechanism?

I think checkpoints are more about preserving task state than memoizing execution results, as it is in caching. And not sure about "targets", not familiar with that concept in Flyte.

I think a good diagram would be helpful for this document.

Thanks!

**Comment by 10sharmashivam** [comment]
> > 
> 
> I think a good diagram would be helpful for this document.
> 
Thanks for the clarification on checkpoints, it helped clear things up. And I would try to create a Data flow diagram for Caching, and add it to [flyteorg/static-resources](https://github.com/flyteorg/static-resources/tree/main/flytesnacks/user_guide) (I believe this is the appropriate location for storing PNG files that can be linked in the documentation—please let me know if that’s correct). 

And once that is approved and merged, will use its raw github user content link to add to caching's documentation.

**Comment by eapolinario** [comment]
> And I would try to create a Data flow diagram for Caching, and add it to [flyteorg/static-resources](https://github.com/flyteorg/static-resources/tree/main/flytesnacks/user_guide) (I believe this is the appropriate location for storing PNG files that can be linked in the documentation—please let me know if that’s correct).


@10sharmashivam , are you working on this diagram?

**Comment by 10sharmashivam** [comment]
Hi @eapolinario, Since the initial PR was merged, I haven’t made further progress on the diagram. However, if there is still value in adding it, I’d be happy to pick it up. Let me know how you’d like to proceed! and please let me know, correct location also for hosting this diagram.

**Comment by davidmirror-ops** [comment]
@10sharmashivam I think visual learners always appreciate a good diagram. Feel free to push it to `static-resources` :)
